openmdao/docs/features/building_blocks/components/eq_constraints_comp.rst

```diff
@@ -0,0 +1,73 @@
+.. index:: EqualityConstraintsComp Example
+
+.. _eq_constraints_comp_feature:
+
+***********************
+EqualityConstraintsComp
+***********************
+
+`EqualityConstraintsComp` is a specialized component that provides a simple way to implement
+equality constraints.
+
+You can add one or more outputs to an `EqualityConstraintsComp` that compute the difference
+between a pair of input values for the purposes of driving the two inputs to equality. It
+computes the output value as:
+
+.. math::
+
+  name_{output} = name_{mult} \times name_{lhs} - name_{rhs}
+
+
+The following inputs and outputs are associated with each output variable.
+
+=========== ======= ====================================================
+Name        I/O     Description
+=========== ======= ====================================================
+{name}      output  output variable
+lhs:{name}  input   left-hand side of difference equation
+rhs:{name}  input   right-hand side of difference equation
+mult:{name} input   left-hand side multiplier of difference equation
+=========== ======= ====================================================
+
+The right-hand side is optional and will default to zero if not connected.
```

Comment:
The wording is a little weird here. "the rhs is optional" isn't quite right. The input always gets created. Its just optional if you want to connect something to it. 

"The default value for the `rhs:{name}` input can be set to  via the `rhs_val` argument. If the rhs_value is fixed (e.g. 0), then the input can be left unconnected. The `lhs:{name}` input must always have something connected to it.

Reply:
changed in EQComp and in original BalanceComp